### PR TITLE
CBL-2976 : Update socketpp and add pinned cert tests

### PR DESCRIPTION
**Pull request by pasin**
* Updated socketpp submodule to pickup the enhance pinned cert implementation.

* Added pinned cert tests in ReplicatorSGTest.cc.

* Updated SG cert and key by replacing the single self-signed cert with 3 cert chain : sg_cert.pem, sg_key.pem, and cert.pem (only leaf cert). The privkey.pem was replaced with sg_key.pem.

* Added scripts/gen_certs.sh that is used for generating the cert chain.

* Made ssl_config.json and config.json compatible with SG 3.0.0

### CODE REVIEW

File: Replicator/tests/ReplicatorSGTest.cc [hunk 1]
```diff
@@ -1280,3 +1280,119 @@ TEST_CASE_METHOD(ReplicatorSGTest, "Auto Purge Enabled(default) - Delete then Cr
...
+        "+4F330aYRvDKDf8r+ve3DtchkUpV9Xa1kcDFyTcYGKBrINtjRmCIblA1fezw59ZT\r\n" \
+        "S5TnM2/TjtQ=\r\n"                                                     \
+        "-----END CERTIFICATE-----\r\n";
+    
+    replicate(kC4OneShot, kC4Disabled, false);
+    CHECK(_callbackStatus.error.domain == NetworkDomain);
+    CHECK(_callbackStatus.error.code == kC4NetErrTLSCertUntrusted);
+}
+
+TEST_CASE_METHOD(ReplicatorSGTest, "Pinned Certificate Success2", "[.SyncServer]") {
```
Comment (borrrden) [inline Replicator/tests/ReplicatorSGTest.cc:1315]: This name does not really describe what this test does.  I can't really tell what its purpose is.  It doesn't seem to pin any certs.

File: Replicator/tests/data/README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ If you want to run a second Sync Gateway instance with SSL, to test SSL connecti
 4. Open another shell at this directory
 5. `sync_gateway ssl_config.json`
 
-This uses a self-signed certificate. You can find a copy of the certificate at `cert.pem`; the private key is `privkey.pem`.
+This uses a self-signed certificate. You can find a copy of the certificate at `sg_cert.pem`; the private key is `privkey.pem`.
```
Comment (borrrden) [inline Replicator/tests/data/README.md:16]: Private key name needs updating too.

**Review by borrrden — CHANGES_REQUESTED** [review]

File: Replicator/tests/ReplicatorSGTest.cc (hunk 1 above)
Comment (pasin) [reply Replicator/tests/ReplicatorSGTest.cc:1315]: Sorry, that test should be removed.

**Comment by pasin** [comment]
PR failure on Windows is not related to this PR.

**Review by jianminzhao — APPROVED** [review]

**Review by borrrden — APPROVED** [review]

**Pull request merged**